packages/core/src/auth/activation.ts

```diff
@@ -3,16 +3,32 @@
  * SPDX-License-Identifier: Apache-2.0
  */
 
+import * as vscode from 'vscode'
 import { Auth } from './auth'
 import { LoginManager } from './deprecated/loginManager'
 import { fromString } from './providers/credentials'
 import { getLogger } from '../shared/logger'
 import { ExtensionUse } from './utils'
-import { isCloud9 } from '../shared/extensionUtilities'
+import { isAmazonQ, isCloud9, isSageMaker } from '../shared/extensionUtilities'
 import { isInDevEnv } from '../shared/vscode/env'
 import { isWeb } from '../shared/extensionGlobals'
+import { CredentialsProviderManager } from './providers/credentialsProviderManager'
+import { SharedCredentialsProviderFactory } from './providers/sharedCredentialsProviderFactory'
+import { Ec2CredentialsProvider } from './providers/ec2CredentialsProvider'
+import { EcsCredentialsProvider } from './providers/ecsCredentialsProvider'
+import { EnvVarsCredentialsProvider } from './providers/envVarsCredentialsProvider'
+
+interface SagemakerCookie {
+    authMode?: string
```

Comment:
What are the possible values? Make this a stricter type than just `string`, e.g.
```
type AuthMode = 'Sso' | 'Iam'
interface SagemakerCookie {
    authMode?: AuthMode
}
```

Reply:
Your assumption is correct. will adjust it